columnflow/tasks/framework/mixins.py

```diff
@@ -1981,24 +1738,46 @@ def resolve_param_values(cls, params):
                     else:
                         multi_var_parts.append(parts)
 
-                # resolve single variables
-                variables = cls.find_config_objects(
-                    single_vars,
-                    config_inst,
-                    od.Variable,
-                    config_inst.x("variable_groups", {}),
-                )
+                def resolve_objects_multi_config(
+                    cls,
+                    objects,
+                    config_insts,
+                    object_cls,
+                    object_groups,
+                ) -> [str]:
+                    outp = None
+                    all_obj = set()
+                    config_obj = dict()
+                    for _config_inst in config_insts:
+                        config_obj[_config_inst] = set(cls.find_config_objects(
+                            objects,
+                            _config_inst,
+                            object_cls,
+                            _config_inst.x(object_groups, {}),
+                        ))
+                        all_obj |= config_obj[_config_inst]
+
+                        if outp:
+                            outp &= config_obj[_config_inst]
+                        else:
+                            outp = config_obj[_config_inst]
+
+                        # warnings for objects not defined in all configs
+                        if len(all_obj - config_obj[_config_inst]) != 0:
+                            for obj in (all_obj - config_obj[_config_inst]):
+                                law.logger.get_logger(cls.task_family).warning(
+                                    f"The object {object_cls} with name {obj} is not defined "
+                                    f"in the config {_config_inst.name}.",
+                                )
+                    return list(outp)
```

Comment:
This was initially the idea behind this function and why it is written in a general way (not specific to od.Variable objects). The reason it is not used for `CategoryMixin` now was because this would be another added overhead and since the amount of categories in an analysis can become quite large it would slow down the task tree creation even more.

Reply:
Yes, I think we should tackle some issues regarding the over-resolving of parameters before adding these cross checks.

I think we started with the variables, because if this is not checked here, it might happen that certain histograms are required for one config that cannot be produced. This cannot happen in the case of categories since categories are not passed to any downstream tasks.
In any case, having variables/categories defined in one config but not another, but still requesting to plot them for multiple configs is a user error that should not really happen. When possible to include them without overhead, I'd be in favor to include cross-checks, but this should not prevent us from merging this PR.

A more prevalent issue we should tackle before merging is the consistency of category ids between multiple configs. We mentioned these issues in the notes in #564 (bullet 4 and 5).

Reply:
> A more prevalent issue we should tackle before merging is the consistency of category ids between multiple configs. We mentioned these issues in the notes in https://github.com/columnflow/columnflow/issues/564 (bullet 4 and 5).

But doesn't this issue go in the same direction? You point a potential slowing down of the resolution of the task tree when using a central implementation to resolve both variables and categories. Do you have a feeling of how large this slow down would be? Since we don't really know at the moment which part of the code actually slows this down, I have the feeling that this claim here is more a guess than a fact. Personally, I really would prefer a central implementation of the resolution of variables/parameters across configs because I fear that if we don't do this now, it will be harder to do later on when the code has evolved even more, meaning that it won't happen (realistically speaking) and the code will be harder to maintain.

Regardless of things slowing down or not, I agree that implementing this resolution every where it could potentially be used might be an unnecessary overhead for this PR. Nevertheless, since I have the feeling that we generally agree that a central implementation does make sense long term, I think already preparing a central implementation here would make it easier to implement in other places later on.

Reply:
OK, I moved this function to the `AnalysisTask` and added some API documentation

Reply:
The issues regarding inconsistent category ids won't be solved via this function because currently, we only compare names of object names, not the content of these objects itself. We discussed this already with @riga  and decided that we do not want to add consistency checks between objects, but of course we can discuss this again in a larger group.